packages/@aws-cdk/aws-glue-alpha/lib/jobs/scala-spark-etl-job.ts

```diff
@@ -8,5 +8,176 @@
+    const profilingMetricsArgs = { '--enable-metrics': '' };
+
+    // Gather executable arguments
+    const execuatbleArgs = this.executableArguments(props);
+
+    // Mandatory className argument
+    if (props.className === undefined) {
+      throw new Error('className must be set for Scala ETL Jobs');
+    }
+
+    // Conbine command line arguments into a single line item
+    const defaultArguments = {
+      ...execuatbleArgs,
+      ...continuousLoggingArgs,
+      ...profilingMetricsArgs,
+      ...sparkUIArgs?.args,
+      ...this.checkNoReservedArgs(props.defaultArguments),
+    };
+
+    if ((!props.workerType && props.numberOrWorkers !== undefined) || (props.workerType && props.numberOrWorkers === undefined)) {
+      throw new Error('Both workerType and numberOrWorkers must be set');
+    }
+
+    const jobResource = new CfnJob(this, 'Resource', {
+      name: props.jobName,
+      description: props.description,
+      role: this.role.roleArn,
+      command: {
+        name: JobType.ETL,
+        scriptLocation: this.codeS3ObjectUrl(props.script),
+      },
+      glueVersion: props.glueVersion ? props.glueVersion : GlueVersion.V4_0,
+      workerType: props.workerType ? props.workerType : WorkerType.G_2X,
+      numberOfWorkers: props.numberOrWorkers ? props.numberOrWorkers : 10,
+      maxRetries: props.maxRetries,
+      executionProperty: props.maxConcurrentRuns ? { maxConcurrentRuns: props.maxConcurrentRuns } : undefined,
+      timeout: props.timeout?.toMinutes(),
+      connections: props.connections ? { connections: props.connections.map((connection) => connection.connectionName) } : undefined,
+      securityConfiguration: props.securityConfiguration?.securityConfigurationName,
+      tags: props.tags,
+      defaultArguments,
+    });
+
+    const resourceName = this.getResourceNameAttribute(jobResource.ref);
+    this.jobArn = this.buildJobArn(this, resourceName);
+    this.jobName = resourceName;
+  }
+
+  /**
+   * Set the executable arguments with best practices enabled by default
+   *
+   * @param props
+   * @returns An array of arguments for Glue to use on execution
+   */
+  private executableArguments(props: ScalaSparkEtlJobProps) {
+    const args: { [key: string]: string } = {};
+    args['--job-language'] = JobLanguage.SCALA;
+    args['--class'] = props.className!;
+
+    if (props.extraJars && props.extraJars?.length > 0) {
+      args['--extra-jars'] = props.extraJars.map(code => this.codeS3ObjectUrl(code)).join(',');
+    }
+
+    return args;
+  }
+
+  private setupSparkUI(role: iam.IRole, sparkUiProps: SparkUIProps) {
+
+    validateSparkUiPrefix(sparkUiProps.prefix);
+    const bucket = sparkUiProps.bucket ?? new Bucket(this, 'SparkUIBucket');
+    bucket.grantReadWrite(role, cleanSparkUiPrefixForGrant(sparkUiProps.prefix));
+    const args = {
+      '--enable-spark-ui': 'true',
+      '--spark-event-logs-path': bucket.s3UrlForObject(sparkUiProps.prefix),
```

Comment:
Don't rollback the fix available on the upstream.
https://github.com/aws/aws-cdk/pull/29357/files

Reply:
Thanks, this is a great call-out

Reply:
This should be addressed in the next PR to this repo. Thanks again for catching this as we are further behind the upstream than we ever intended to be.